### PR TITLE
Aws cloudwatch

### PR DESCRIPTION
**Pull request by DayoOliyide**
Trackit can now push metrics to cloudwatch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -694,6 +695,45 @@ This function will return `true` only when the type is one of the listed types.
...
+
+
+```clojure
+(import 'java.util.concurrent.TimeUnit)
+(import 'com.amazonaws.services.cloudwatch.AmazonCloudWatchAsyncClientBuilder)
+
+(start-reporting!
+   {:type                        :cloudwatch
+    ;; The custom namespace in cloudwatch under which metrics will be published
+    :namespace                   "trackit-namespace"
```
Comment (BrunoBonacci) [inline README.md]:
```suggestion
    :namespace                   "trackit"
```
For uniformity, all other reporters use a `prefix` called `trackit`.
could you please use the same here?

File: README.md [hunk 2]
```diff
@@ -694,6 +695,45 @@ This function will return `true` only when the type is one of the listed types.
...
+    ;; If present, the aws async client used to report push metrics. Most users don't need to
+    ;; provide this, but the option is provided for those who might need it
+    :async-client                (.build (AmazonCloudWatchAsyncClientBuilder/standard))
```
Comment (BrunoBonacci) [inline README.md]: Does make sense to expose this option? is any other alternative?

File: trackit-cloudwatch/src/samsara/trackit/reporter_cloudwatch.clj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  (:import  [java.util.concurrent TimeUnit]
+            [com.amazonaws.regions Regions]
+            [com.amazonaws.services.cloudwatch AmazonCloudWatchAsyncClientBuilder]
+            [io.github.azagniotov.metrics.reporter.cloudwatch CloudWatchReporter]))
+
+(defn start-reporting
+  [registry
+   {:keys [namespace async-client reporting-frequency-seconds rate-unit duration-unit
+           dry-run]
+    :or   {async-client                (.build (AmazonCloudWatchAsyncClientBuilder/standard))
```
Comment (BrunoBonacci) [inline trackit-cloudwatch/src/samsara/trackit/reporter_cloudwatch.clj:16]: again, not too sure it useful to have this as a configurable option

**Review by BrunoBonacci — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (DayoOliyide) [reply README.md]: I'll change the :namespace value. However, to my knowledge prefix doesn't really exist in 
cloudwatch metrics world. It's mainly namespace, metric and dimensions. I don't see how prefix will translate.

File: trackit-cloudwatch/src/samsara/trackit/reporter_cloudwatch.clj (hunk 3 above)
Comment (DayoOliyide) [reply trackit-cloudwatch/src/samsara/trackit/reporter_cloudwatch.clj:16]: I've removed the async-client from the documentation and the start-reporting! multimethod.
I would still want to keep it as an option in the cloudwatch reporter, as I've come across 
an issue where the apparent solution was to begin a subsegment for cloudwatch before each
call to cloudwatch.
There may be a better solution, but I'd like to keep this option open until I find it.

**Comment by BrunoBonacci** [comment]
released with`[samsara/trackit-all "0.8.0"]`

**Pull request merged**